metacat-common-server/src/main/java/com/facebook/presto/plugin/ColumnDetailHandle.java

```diff
@@ -135,46 +146,46 @@ public Boolean getIsIndexKey() {
         return isIndexKey;
     }
 
-    public ColumnMetadata getColumnMetadata()
-    {
-        StringBuilder comments = new StringBuilder(comment==null?"":comment).append(" ")
-                .append("nullable=").append(isNullable).append(", ")
-                .append("columnLength=").append(size).append(", ")
-                .append("default=").append(defaultValue).append(", ")
-                .append("sortKey=").append(isSortKey).append(", ")
-                .append("indexKey=").append(isIndexKey);
+    /**
+     * Returns the column metadata.
+     * @return column metadata
+     */
+    public ColumnMetadata getColumnMetadata() {
+        final StringBuilder comments = new StringBuilder(comment == null ? "" : comment).append(" ")
+            .append("nullable=").append(isNullable).append(", ")
+            .append("columnLength=").append(size).append(", ")
+            .append("default=").append(defaultValue).append(", ")
+            .append("sortKey=").append(isSortKey).append(", ")
+            .append("indexKey=").append(isIndexKey);
 
-        return new ColumnDetailMetadata(columnName, columnType, isPartitionKey==null?false:isPartitionKey,
-                comments.toString(), false, sourceType, size, isNullable, defaultValue, isSortKey, isIndexKey);
+        return new ColumnDetailMetadata(columnName, columnType, isPartitionKey == null || isPartitionKey,
+            comments.toString(), false, sourceType, size, isNullable, defaultValue, isSortKey, isIndexKey);
     }
```

Comment:
If isPartitionKey == null, the previous one return false. the changed one return true.